ui/app/adapters/pki/sign-intermediate.js

```diff
@@ -0,0 +1,19 @@
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+import ApplicationAdapter from '../application';
+
+export default class PkiSignIntermediateAdapter extends ApplicationAdapter {
+  namespace = 'v1';
+
+  createRecord(store, type, snapshot) {
```

Comment:
for cross-signing I added a `sign-intermediate` case to the `urlForCreateRecord` method - that could be used here unless extending adapters just makes things way too complicated.

Since you made a separate model, separation of concerns here is nice!

Reply:
I needed to add the id manually to the response, that's why I have it separate. Let's keep it that way for now, and maybe in our cleanup/documentation phase we can consolidate/extend